GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -452,10 +454,13 @@ + (GIDSignIn *)sharedInstance {
 
 - (id)initPrivate {
   GIDKeychainHandler *keychainHandler = [[GIDKeychainHandler alloc] init];
-  return [self initWithKeychainHandler:keychainHandler];
+  GIDDataFetcher *dataFetcher = [[GIDDataFetcher alloc] init];
```

Comment:
You can use `id<GIDDataFetcher>` on the left hand side because the initializer below is expecting `id<GIDDataFetcher>` type, and we should prefer protocol to class wherever possible.

Same for `GIDKeychainHandler` above.